manifests/directory.pp

```diff
@@ -0,0 +1,46 @@
+# Define: patch::file
+#
+# Apply a patch on a directory.
+#
+# === Authors
+#
+# Martin Meinhold <Martin.Meinhold@gmx.de>
+#
+# === Copyright
+#
+# Copyright 2014 Martin Meinhold, unless otherwise noted.
+#
+define patch::directory (
+  $cwd,
```

Comment:
What about using the resource title by default as `cwd`? I imaging this would simpify the usage as otherwise you'd have to repeat the directory name in the resource title and for the `cwd`attribute. The example in the readme would then look like 

```
patch::directory { '/path/to/target/directory':
  diff_source => '/path/to/diff',
  strip       => 2,
}
```

Otherwise the `title` is just used to generate a unique patch filename.